staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics_test.go

```diff
@@ -226,6 +226,48 @@ func TestStorageSizeCollector(t *testing.T) {
 
 }
 
+func TestUpdateObjectCount(t *testing.T) {
+	registry := metrics.NewKubeRegistry()
+	registry.Register(objectCounts)
+	testedMetrics := "apiserver_storage_objects"
+
+	testCases := []struct {
+		desc     string
+		resource string
+		count    int64
+		want     string
+	}{
+		{
+			desc:     "successful fetch",
+			resource: "foo",
+			count:    10,
+			want: `# HELP apiserver_storage_objects [STABLE] Number of stored objects at the time of last check split by kind. In case of a fetching error, the value will be -1.
+# TYPE apiserver_storage_objects gauge
+apiserver_storage_objects{resource="foo"} 10
+`,
+		},
+		{
+			desc:     "failed fetch",
+			resource: "bar",
+			count:    -1,
+			want: `# HELP apiserver_storage_objects [STABLE] Number of stored objects at the time of last check split by kind. In case of a fetching error, the value will be -1.
+# TYPE apiserver_storage_objects gauge
+apiserver_storage_objects{resource="bar"} -1
+`,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.desc, func(t *testing.T) {
+			defer registry.Reset()
+			UpdateObjectCount(test.resource, test.count)
+			if err := testutil.GatherAndCompare(registry, strings.NewReader(test.want), testedMetrics); err != nil {
```

Comment:
@dgrisonnet - seems like something we want to fix there ^^

Reply:
Yeah that's definitely something that should be fixed. Great find @machine424 :+1: 

The testutil function we have is just a wrapper around the function in client_golang, so we would have to fix it there: https://github.com/prometheus/client_golang/blob/1bae6c1e6314f6a20be183a7277059630780232a/prometheus/testutil/testutil.go#L201-L203

Reply:
Cool, now that we agree that this can be improved, I can see with client_golang folks (I think this was introduced here https://github.com/prometheus/client_golang/pull/1143/files), independently of this PR.

Reply:
Created an issue https://github.com/prometheus/client_golang/issues/1351

Reply:
As I said the issue https://github.com/prometheus/client_golang/issues/1351 is not a prerequisite, we can try to merge this without.